first, initial release (#1)
adfinis-sygroup/back8sup: README.md

# back8sup

A simple way to backup your kubernetes resources

> ## Deployment w/ HELM v3.x

Reviewer: ```suggestion
## Deployment with Helm
```

Reviewer: Helm is not an acronym.

```

> helm install back8sup charts/back8sup

Reviewer: ```suggestion
helm repo add adfinis https://charts.adfinis.com
helm install back8sup adfinis/back8sup
```

```
## Configuration

> Configuration is done via environment variables inside the container and an additional configmap

Reviewer: ```suggestion
Back8sup is configured via environment variables inside the container and an additional configmap.
```

### Environment variables

```
API_ENDPOINT=${API_ENDPOINT:-https://kubernetes.local:6443}
CA_CERT=${CA_CERT:-/etc/ssl/ca.crt}
TOKEN_FILE=${TOKEN_FILE:-/var/run/secrets/sa}
DST_FOLDER=${DST_FOLDER:-/mnt/back8sup}
CONFIGMAP_PATH=${CONFIGMAP_PATH:-/etc/config.yaml}
EXPORT_FORMAT=${EXPORT_FORMAT:-yaml}
```

### ConfigMap

> 

Reviewer: This should probably mention that the configmap is managed by Helm (At least I'm assuming it will be).

Author: ```suggestion
(managed by Helm)
```

Reviewer: I'm a fan of full sentences.

Author: better?

# back8sup

A simple way to backup your kubernetes resources

## Deployment w/ HELM v3.x

```
helm repo add adfinis https://charts.adfinis.com
helm install back8sup adfinis/back8sup
```
## Configuration

Back8sup is configured via environment variables inside the container and an additional configmap.

### Environment variables

```
API_ENDPOINT=${API_ENDPOINT:-https://kubernetes.local:6443}
CA_CERT=${CA_CERT:-/etc/ssl/ca.crt}
TOKEN_FILE=${TOKEN_FILE:-/var/run/secrets/sa}
DST_FOLDER=${DST_FOLDER:-/mnt/back8sup}
CONFIGMAP_PATH=${CONFIGMAP_PATH:-/etc/config.yaml}
EXPORT_FORMAT=${EXPORT_FORMAT:-yaml}
```

### ConfigMap

> The configmap is managed by helm and configured in the `vales.yaml`

Reviewer: ```suggestion
The configmap is managed by helm and configured in the `values.yaml` file.
```

# back8sup

A simple way to backup your kubernetes resources

## Why?

In some cases, using Velero with an S3 storage provider is an overkill. 

This script keeps things very basic and simple, and makes it easy to backup your Kubernetes metadata quickly by using your default StorageClass.

## Deployment with Helm

```
helm repo add adfinis https://charts.adfinis.com
helm install back8sup adfinis/back8sup
```
## Configuration

Back8sup is configured via environment variables inside the container and an additional configmap.

### Environment variables

```
API_ENDPOINT=${API_ENDPOINT:-https://kubernetes.local:6443}
CA_CERT=${CA_CERT:-/etc/ssl/ca.crt}
TOKEN_FILE=${TOKEN_FILE:-/var/run/secrets/sa}
DST_FOLDER=${DST_FOLDER:-/mnt/back8sup}
CONFIGMAP_PATH=${CONFIGMAP_PATH:-/etc/config.yaml}
EXPORT_FORMAT=${EXPORT_FORMAT:-yaml}
```

### ConfigMap

The configmap is managed by helm and configured in the `vales.yaml`

> Example configuration (see `src/config.sample.yaml`)

Reviewer: ```suggestion
Example configuration (see [`src/config.sample.yaml`](src/config.sample.yaml))
```
